aptos-move/framework/aptos-framework/sources/staking_contract.move

```diff
@@ -346,6 +347,27 @@ module aptos_framework::staking_contract {
         emit_event(&mut store.reset_lockup_events, ResetLockupEvent { operator, pool_address });
     }
 
+    public entry fun update_commision(staker: &signer, operator: address, new_commission_percentage: u64) acquires Store {
+        assert!(
+            new_commission_percentage >= 0 && new_commission_percentage <= 100,
+            error::invalid_argument(EINVALID_COMMISSION_PERCENTAGE),
+        );
+
+        let staker_address = signer::address_of(staker);
+        assert!(exists<Store>(staker_address), error::not_found(ENO_STAKING_CONTRACT_FOUND_FOR_STAKER));
+
+        let store = borrow_global_mut<Store>(staker_address);
+        let staking_contract = simple_map::borrow_mut(&mut store.staking_contracts, &operator);
+        distribute_internal(staker_address, operator, staking_contract, &mut store.distribute_events);
+        request_commission_internal(
+            operator,
+            staking_contract,
+            &mut store.add_distribution_events,
+            &mut store.request_commission_events,
+        );
+        staking_contract.commission_percentage = new_commission_percentage;
```

Comment:
We should emit an event here. Since we can't add fields to existing struct (for compat reasons), we'll need to create a new resource. We also might want to put it into some kind of resource groups so if we add more new events in the future (and thus new resources), they can all belong to the same resource group so new storage slots are not created.

Example: https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/move-examples/resource_groups/primary/sources/primary.move#L6 (we'd not want global scope). You can also check with @davidiw for guidance (he needs to write docs for resource groups anyway)